unified-codes/apps/web/src/sagas/DetailSaga.ts

```diff
@@ -0,0 +1,112 @@
+import { call, put, takeEvery, all, delay } from 'redux-saga/effects';
+
+import {
+  AlertSeverity,
+  IEntity,
+  IAlert,
+} from '@unified-codes/data';
+
+import {
+  DETAIL_ACTIONS,
+  DetailActions,
+  AlertActions,
+  IDetailFetchEntityAction,
+} from '../actions';
+
+
+const ALERT_SEVERITY = {
+  FETCH: AlertSeverity.info,
+  ERROR: AlertSeverity.error,
+};
+
+const ALERT_TEXT = {
+  FETCH: 'Fetching...',
+  ERROR: 'Could not fetch entity.',
+};
+
+const alertError: IAlert = {
+  isVisible: true,
+  severity: ALERT_SEVERITY.ERROR,
+  text: ALERT_TEXT.ERROR,
+};
+
+const alertFetch: IAlert = {
+  isVisible: true,
+  severity: ALERT_SEVERITY.FETCH,
+  text: ALERT_TEXT.FETCH,
+};
+
+const getEntityQuery = (code: string) => `
+{
+  entity (code: "${code}") {
+    code
+    description
+    type
+    has_property {
+      type
+      value
+    }
+    # form_category
+    has_child {
+      code
+      description
+      type
+      # form
+      has_child {
+        code
+        description
+        type
+        # unit_of_use
```

Comment:
I think discussed this with @mark-prins. Not necessarily a problem (for now), but I think this may be complicated as this only works if we assume we will always be fetching `Drug` entities (as other entities may not have forms etc.)?

Reply:
Currently if there are no forms it will just fetch whatever child happens to be there (so yeah the comment is probably a little misleading - oops). E.g. for `Other` this will just be `unit_of_use` with the current test data. I did a bit of testing also, if there are no children it just returns a null object for the child. 

Agree this is not very nice on the client-side as we start having to read types and things to parse the data since everything is an `Entity`. Entities for everyone 🥳

Reply:
I would `recurse` to a depth of 4, as that would pick up the properties on the unit of use as well.

Reply:
Hmm, I'm not sure how to do this (`recurse`) in pure GraphQL (though it's possible in DgraphQL). Would adding a `has_property` field to each of the children provide a sufficient level of detail? Or do we also want children of `unit_of_use`?

Reply:
oh, right. ignore the comment then, sorry! merge ahoy!